chef/automate
# Fix API docs' paths

### :nut_and_bolt: Description: What code changed, and why?

To make the API docs match the paths used in the system, this change makes all proto files use their complete paths, i.e., `/api/v0/foo/bar`, not `/foo/bar`. The rest of the PR is fixing the fallout.

#### Notes

- [x] authz introspection makes use of the proto's paths, and they'll change now
  - there's two ways to fix this: massage the paths, or make it work with the fixed paths
  - earlier, we have made exceptions from API compatibility for introspection, and I believe it's warranted to ignore it here, too; if working with the fixed paths is simpler in any way (which I think it'll be)
- [x] for the other paths, `/api/v0`, we'll need to pick either `/api/v0` or `/api/v1`, as the primary path -- the one used in the proto annotations that we process
  - there's an additional path setting, I think, I'll try to see if we can leverage that, or if it hinders doc generation -- **update** I've decided against that, since updating the protos programmatically (read: something better than sed) is blocked. So, I'll make the proto paths say `/api/v0` and use the load-balancer or the gateway to massage `/api/v1`s to `/api/v0`.
  - [x] for introspection, only the primary path will matter, so maybe we'll have to rewrite some paths in the process (UI or backend) -- **still needs fixing**
- [x] fix remaining test fallout
- [x] cleanup the git history, this should not contain changes in the automate-chef-io api docs data directory

### :chains: Related Resources

- #3419 👈 this incorporates that work (rebased!)
- #3401 👈 this is the issue

### :+1: Definition of Done

- our docs have the correct API docs
- introspection from the browser still works

### :athletic_shoe: How to Build and Test the Change

- To see that the UI still behaves, rebuild:
  - automate-load-balancer
  - automate-gateway
  - automate-ui

To view the docs:
  - go to `components/automate-chef-io`
  - run `make sync_swagger_files` to regenerate needed files
  - run `make serve` to start the docs server
  - browse to `http://localhost:1313/docs/api`

For the public API, nothing has changed. Note the passing tests 😉 

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
